R/utils.R

```diff
@@ -565,234 +579,194 @@ bootstrap_is_cleaned_up <- function(.boot_run){
-    verbose_msg("Bootstran run has not been set up")
-    return(invisible(TRUE))
-  }
-  boot_models <- get_boot_models(.mod)
-  output_dirs <- purrr::map_chr(boot_models, get_output_dir, .check_exists = FALSE)
-  if(!any(fs::dir_exists(output_dirs))){
-    verbose_msg("Bootstran run has not been submitted")
-    return(invisible(TRUE))
-  }
-  models_finished <- map_lgl(boot_models, ~nonmem_finished_impl(.x))
-  return(all(models_finished))
-}
 
-#' @describeIn check_nonmem_finished takes a `list` of `bbi_nonmem_model` objects.
 #' @export
 check_nonmem_finished.list <- function(.mod, ...) {
-  assert_list(.mod)
-  check_model_object_list(.mod, .mod_types = c(NM_MOD_CLASS))
-  output_dirs <- purrr::map_chr(.mod, get_output_dir, .check_exists = FALSE)
-  if(!any(fs::dir_exists(output_dirs))){
-    verbose_msg("Models have not been submitted")
-    return(invisible(TRUE))
-  }
+  check_model_object_list(.mod, .mod_types = c(NM_MOD_CLASS, NMBOOT_MOD_CLASS))
   # Return logical values as vector (unique handling)
   #  - used in `wait_for_nonmem`
-  models_finished <- map_lgl(.mod, ~nonmem_finished_impl(.x))
+  models_finished <- map_lgl(.mod, ~model_is_finished(.x))
   return(models_finished)
 }
 
 
-nonmem_finished_impl <- function(.mod){
-  mod_path <- build_path_from_model(.mod, ".lst")
 
-  # look for model to be finished and then test output
-  model_finished <- if(file.exists(mod_path)){
-    read_lines(mod_path) %>% str_detect("Stop Time") %>% any()
+#' @describeIn model_status Returns messages indicating which model(s) have
+#'  finished executing and which are still running.
+#' @param max_print max number of models to explicitly print to the console. If
+#' the number of finished or incomplete models are greater than this number,
+#' just print the _number_ of models.
+#'
+#' @export
+get_model_status <- function(.mod, max_print = 10, ...){
+  UseMethod("get_model_status")
+}
+
+
+#' @export
+get_model_status.default <- function(.mod, max_print = 10, ...){
+  checkmate::assert_number(max_print, lower = 1)
+
+  # Coerce to list of models regardless of input
+  mod_list <- mod_list_setup(.mod)
+  # Exit for bootstrap runs that haven't been set up or were already cleaned up
+  if(is.null(mod_list)) return(invisible(NULL))
+
+  mod_ids <- purrr::map_chr(mod_list, get_model_id)
+  res <- check_nonmem_finished(mod_list) %>% tibble::as_tibble() %>%
+    dplyr::transmute(model_id = mod_ids, finished = .data$value)
+
+  res_fin <- res$model_id[res$finished]
+  if(length(res_fin) > 0 && length(res_fin) <= max_print){
+    mods_fin <- paste(res_fin, collapse = ", ")
+    message(glue("\n The following model(s) have finished: `{mods_fin}`"))
+  }else{
+    message(glue("\n{length(res_fin)} model(s) have finished"))
+  }
+
+  res_inc <- res$model_id[!res$finished]
+  if(length(res_inc) > 0 && length(res_inc) <= max_print){
+    mods_inc <- paste(res_inc, collapse = ", ")
+    message(glue("\n The following model(s) are incomplete: `{mods_inc}`"))
```

Comment:
I think this (and the message on 723) should probably say "...are incomplete or have not yet been run"

This may be a nitpick, but to me this implies that those models are all _running_ but not yet complete (which is inaccurate). Especially because  `bbi_nonmem_model_status` (our "source of truth") does distinguish between those two statuses. Thoughts?

Reply:
If you agree, this would require changing the expected strings in several tests.